generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -108,20 +108,42 @@ private FieldValueSource getNumericSource(FieldSpec fieldSpec) {
             ? new NumericRestrictions()
             : fieldSpec.getNumericRestrictions();
 
-        int numericScale = fieldSpec.getGranularityRestrictions() != null
-            ? fieldSpec.getGranularityRestrictions().getNumericScale()
-            : 0;
+        int numericScale = getNumericScale(restrictions, fieldSpec);
 
-        if (numericScale == 0) {
+        if ((restrictions.min == null || restrictions.max == null) || isIntegerFieldValue(restrictions, numericScale)) {
             return new IntegerFieldValueSource(
                 restrictions,
                 getBlacklist(fieldSpec));
-        } else {
-            return new RealNumberFieldValueSource(
-                restrictions,
-                getBlacklist(fieldSpec),
-                numericScale);
         }
+
+        return new RealNumberFieldValueSource(
+            restrictions,
+            getBlacklist(fieldSpec),
+            numericScale);
+    }
+
+    private int getNumericScale(NumericRestrictions numericRestrictions, FieldSpec fieldSpec) {
+        if (fieldSpec.getGranularityRestrictions() != null) {
+            return fieldSpec.getGranularityRestrictions().getNumericScale();
+        }
+
+        if (numericRestrictions.min != null && numericRestrictions.max != null) {
+            if (numericRestrictions.min.getLimit().scale() > numericRestrictions.max.getLimit().scale()) {
+                return numericRestrictions.min.getLimit().scale();
+            }
+
+            return numericRestrictions.max.getLimit().scale();
+        }
+
+        return 0;
+    }
+
+    private boolean isIntegerFieldValue(NumericRestrictions numericRestrictions, int numericScale) {
```

Comment:
This might seem picky 👀, but i think it would read easier as
isFieldValueAnInteger
as it currently reads like there is an 'integerFieldValue' class of some sort